hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ContractBurnHTSV2SecurityModelSuite.java

```diff
@@ -0,0 +1,693 @@
+                                .signedBy(TOKEN_TREASURY)
+                                .payingWith(TOKEN_TREASURY)
+                                // Verify that the top level status of the transaction is CONTRACT_REVERT_EXECUTED
+                                .hasKnownStatus(CONTRACT_REVERT_EXECUTED),
+                        // Assert that the token is NOT burned
+                        getTokenInfo(NON_FUNGIBLE_TOKEN).hasTotalSupply(1L),
+                        // Assert the token is NOT burned from the token treasury account
+                        getAccountBalance(TOKEN_TREASURY).hasTokenBalance(NON_FUNGIBLE_TOKEN, 1L),
+                        // Create a key with thresh 1/2 with sigs:  new ed25519 key, contractId of
+                        // BURN_TOKEN_VIA_DELEGATE_CALL contract
+                        newKeyNamed(TRESHOLD_KEY_CORRECT_CONTRACT_ID)
+                                .shape(TRESHOLD_KEY_SHAPE.signedWith(sigs(ON, MIXED_BURN_TOKEN))),
+                        // Update the token's supply to have the threshold key wit the wrong contract id
+                        tokenUpdate(NON_FUNGIBLE_TOKEN).supplyKey(TRESHOLD_KEY_CORRECT_CONTRACT_ID),
+                        // Update the signer of the transaction to have the threshold key with the wrong contract id
+                        cryptoUpdate(SIGNER).key(TRESHOLD_KEY_CORRECT_CONTRACT_ID),
+                        // Test Case 2: A Signer paying and signing a NON FUNGIBLE token burn transaction,
+                        // SIGNER → call → CONTRACT → delegatecall → PRECOMPILE
+                        // The token and the signer have updated keys
+                        contractCall(
+                                MIXED_BURN_TOKEN,
+                                "burnTokenDelegateCall",
+                                BigInteger.valueOf(0),
+                                HapiParserUtil.asHeadlongAddress(
+                                        asAddress(spec.registry().getTokenID(NON_FUNGIBLE_TOKEN))),
+                                serialNumber1)
+                                .via(DELEGATE_CALL_WHEN_NON_FUNGIBLE_TOKEN_HAS_CONTRACT_ID_SIGNER_SIGNS)
+                                .gas(GAS_TO_OFFER)
+                                .signedBy(SIGNER)
+                                .payingWith(SIGNER)
+                                // Verify that the top level status of the transaction is CONTRACT_REVERT_EXECUTED
+                                .hasKnownStatus(CONTRACT_REVERT_EXECUTED),
+                        // Assert that the token is NOT burned
+                        getTokenInfo(NON_FUNGIBLE_TOKEN).hasTotalSupply(1L),
+                        // Assert the token is NOT burned from the token treasury account
+                        getAccountBalance(TOKEN_TREASURY).hasTokenBalance(NON_FUNGIBLE_TOKEN, 1L),
+                        tokenUpdate(NON_FUNGIBLE_TOKEN).supplyKey(TRESHOLD_KEY_CORRECT_CONTRACT_ID)
```

Comment:
I believe this tokenUpdate is not necessary and we can delete the line

Reply:
fixed